lib/pure/asyncmacro.nim

```diff
@@ -447,8 +325,28 @@ macro multisync*(prc: untyped): untyped =
   result = newStmtList()
   result.add(asyncSingleProc(asyncPrc))
   result.add(sync)
-
-proc await*[T](x: T) =
-  ## The 'await' keyword is also defined here for technical
-  ## reasons. (Generic symbol lookup prepass.)
-  {.error: "Await only available within .async".}
+  # echo result.repr
+
+# overload for type which is not Future[T], based on the yglukhov's patch to chronos: https://github.com/status-im/nim-chronos/pull/47
+template await*(f: untyped): untyped =
```

Comment:
This merely happens to work, I think you need ``f: typed`` here.

Reply:
hm,ok, i will try

Reply:
changed: i'll wait for the tests